rules/no-ignore-file.js

```diff
@@ -0,0 +1,37 @@
+const ignoreFileRE = /^\s*istanbul\s+ignore\s+file(?=\W|$)/u;
+
+module.exports = {
+	meta: {
+		docs: {
+			category: 'Best Practices',
+			description:
+				'This rule raises a warning about "istanbul ignore file" comments.',
+			recommended: false
+		},
+		messages: {
+			ignoreFile: 'Ignoring coverage for whole files is not allowed'
```

Comment:
I'd like to see this somehow specify that it applies to hints only, this makes it sound like setting `exclude` in .nycrc is forbidden.  I'm not great at wording so I'm not sure exactly what I'm looking for, if you can't think of a way to improve this don't consider this a blocker.

Reply:
huh I am not too good with phrasing, `"Manual ignore for non excluded files is not allowed"` is the only thing I can come up with atm. Not sure if this sounds any better. Specially that not everyone uses `.nycrc` config explicitly. Jest allows to set file patterns from which the coverage is gathered.

Reply:
@AndrewFinlay @bcoe @JaKXz anyone have thoughts on this?

Reply:
It looks to me that you'd want this to show up if there was a preferred way to exclude a file that you didn't know about or as a warning that you'd left a file excluded.  In case two you'd just remove the `ignore` so that leaves it as a suggestion to use a preferred method of excluding files, so something like:

`Prefer Istanbul config for ignoring files`

Reply:
`Unexpected file coverage disabling. Prefer ignoring complete files via config`.

Reply:
I like @AndrewFinlay's suggestion with a slight tweak.
```suggestion
			ignoreFile: 'Prefer config for ignoring files'
```

I'm dropping the word `Istanbul` from this message as I don't want jest users to try creating an `.nycrc` (jest doesn't read that file).